### ISSUE TITLE
[8.0-preview.5] Possible Issue with AsyncLocal<T>/TaskCreationOptions.RunContinuationsAsynchronously

### ISSUE THREAD

**Issue by martincostello** [dotnet/runtime#87628]
### Description

I have a number of applications that were targeting .NET 8 preview 4 that have been upgraded to .NET preview 5 this week. In four different applications (only 3 are public), I've observed issues with their UI tests that use [Microsoft.Playwright](https://github.com/microsoft/playwright-dotnet) where a `NullReferenceException` is thrown from within the Playwright code during test execution.

- https://github.com/martincostello/dependabot-helper/pull/690
- https://github.com/martincostello/apple-fitness-workout-mapper/pull/801
- https://github.com/martincostello/alexa-london-travel-site/pull/1752

These failures occur on Windows, macOS and Linux, but _exactly_ what fails seems to vary _slightly_ between the different applications and operating systems.

After doing various investigation, including debugging one of the applications consuming Microsoft.Playwright from source, I wonder if there's an issue somewhere related to `AsyncLocal<T>` and/or `TaskCreationOptions.RunContinuationsAsynchronously` in .NET 8 preview 5? This is partly just a hunch as I've seen PRs into the repo recently around Task-related refactoring, I might be way off in my assessment.

The code for the relevant parts of Playwright in v1.35.0 using these are here:

- [`TaskCreationOptions.RunContinuationsAsynchronously` being used](https://github.com/microsoft/playwright-dotnet/blob/ca7b02ac910e573666f48d502bfabb1c17639e68/src/Playwright/Transport/Connection.cs#L135)
- Code where the `NullReferenceException` comes from, which is [accessing an `AsyncLocal<T>`](https://github.com/microsoft/playwright-dotnet/blob/ca7b02ac910e573666f48d502bfabb1c17639e68/src/Playwright/Transport/Connection.cs#L150)
- [`try-finally` block awaiting a delegate](https://github.com/microsoft/playwright-dotnet/blob/ca7b02ac910e573666f48d502bfabb1c17639e68/src/Playwright/Transport/Connection.cs#L507-L518) that is invoking the `InnerSendMessageToServerAsync()` method containing the above code

When referencing the Playwright code from source I added the following lines before the `AsyncLocal<T>` is accessed:

```diff
if (ApiZone is null)
{
    throw new InvalidOperationException("ApiZone is null.");
}
if (ApiZone.Value is null)
{
    throw new InvalidOperationException("ApiZone.Value is null.");
}
if (ApiZone.Value[0] is null)
{
    throw new InvalidOperationException("ApiZone.Value[0] is null.");
}
```

Re-running the tests with this edit changes the test failures to the following:

```
  Error Message:
   System.InvalidOperationException : ApiZone.Value[0] is null.
  Stack Trace:
     at Microsoft.Playwright.Transport.Connection.InnerSendMessageToServerAsync[T](String guid, String method, Dictionary`2 dictionary, Boolean keepNulls) in C:\Coding\microsoft\playwright-dotnet\src\Playwright\Transport\Connection.cs:line 161
```

This seems to suggest to me that the `try-catch` which is editing `ApiZone.Value[0]` is not running in the execution order that is expected by the async calls. This is just my inference though at trying to explain away the exception - I'm not familiar with the inner workings of Playwright's implementation.

Observing the UI tests with headless mode disabled (as well as reviewing screenshots and recordings) don't seem to suggest any issues with the applications themselves in terms of their expected behaviour (i.e. genuine test failures), it seems that something is just causing Playwright to fail internally.

An additional factor to me thinking this is something related to `AsyncLocal<T>` is that if I run an affected test which is data-driven and remove all of the browser combinations and just run a single browser (whatever that browser is), then the test passed on each of my attempts. Similarly, the same test may fail on different lines during the assertions, rather than always the exact same failure.

### Reproduction Steps

1. Clone https://github.com/martincostello/dependabot-helper/pull/690/commits/60069687a634498d93d293a56fc713aa2ff8e726
2. Run `build.ps1 -TestFilter Can_Configure_Repositories`

### Expected behavior

The tests pass.

### Actual behavior

One or more tests fail with an exception similar to the below:

```
  Failed MartinCostello.DependabotHelper.UITests.Can_Configure_Repositories(browserType: "firefox", browserChannel: null) [6 s]
  Error Message:
   System.NullReferenceException : Object reference not set to an instance of an object.
  Stack Trace:
     at Microsoft.Playwright.Transport.Connection.InnerSendMessageToServerAsync[T](String guid, String method, Dictionary`2 dictionary, Boolean keepNulls) in /_/src/Playwright/Transport/Connection.cs:line 150
   at Microsoft.Playwright.Transport.Connection.WrapApiCallAsync[T](Func`1 action, Boolean isInternal)
   at Microsoft.Playwright.Core.ElementHandle.QuerySelectorAsync(String selector) in /_/src/Playwright/Core/ElementHandle.cs:line 181
   at MartinCostello.DependabotHelper.Pages.ConfigurePage.RepositoryItem.IsSelectedAsync() in /_/tests/DependabotHelper.Tests/Pages/ConfigurePage.cs:line 141
   at Shouldly.ShouldlyTaskExtensions.ShouldBeFalse(Task`1 task) in /_/tests/DependabotHelper.Tests/Infrastructure/ShouldlyTaskExtensions.cs:line 22
   at MartinCostello.DependabotHelper.UITests.<>c__DisplayClass5_0.<<Can_Configure_Repositories>b__0>d.MoveNext() in /_/tests/DependabotHelper.Tests/UITests.cs:line 166
--- End of stack trace from previous location ---
   at MartinCostello.DependabotHelper.Infrastructure.BrowserFixture.WithPageAsync(Func`2 action, String testName) in /_/tests/DependabotHelper.Tests/Infrastructure/BrowserFixture.cs:line 59
   at MartinCostello.DependabotHelper.Infrastructure.BrowserFixture.WithPageAsync(Func`2 action, String testName) in /_/tests/DependabotHelper.Tests/Infrastructure/BrowserFixture.cs:line 64
   at MartinCostello.DependabotHelper.Infrastructure.BrowserFixture.WithPageAsync(Func`2 action, String testName) in /_/tests/DependabotHelper.Tests/Infrastructure/BrowserFixture.cs:line 76
   at MartinCostello.DependabotHelper.Infrastructure.BrowserFixture.WithPageAsync(Func`2 action, String testName) in /_/tests/DependabotHelper.Tests/Infrastructure/BrowserFixture.cs:line 76
   at MartinCostello.DependabotHelper.Infrastructure.BrowserFixture.WithPageAsync(Func`2 action, String testName) in /_/tests/DependabotHelper.Tests/Infrastructure/BrowserFixture.cs:line 76
   at MartinCostello.DependabotHelper.UITests.Can_Configure_Repositories(String browserType, String browserChannel) in /_/tests/DependabotHelper.Tests/UITests.cs:line 126
```

### Regression?

Yes. The test pass with .NET 7.0.7 and with .NET 8 preview 4.

### Known Workarounds

None.

### Configuration

- .NET SDK `8.0.100-preview.5.23303.2`
- Observed on Windows, Linux and macOS GitHub Actions runners as well as locally with Windows 11.

**Comment by martincostello** [comment]
I've updated all 4 applications to disable PGO and now all the tests pass as expected, so looks like Dynamic PGO was indeed the contributing factor. 🕵️

**Comment by AndyAyersMS** [comment]
Ok, thanks for checking. 

Since you have a repro we can try it shouldn't be too hard to track this one down.

**Comment by AndyAyersMS** [comment]
@dotnet/jit-contrib anyone want to dig in? If not, I will take a look when I have time.

**Comment by AndyAyersMS** [comment]
> ### Reproduction Steps
> 1. Clone [martincostello/dependabot-helper@6006968](https://github.com/martincostello/dependabot-helper/commit/60069687a634498d93d293a56fc713aa2ff8e726)
> 2. Run `build.ps1 -TestFilter Can_Configure_Repositories`

Is this an orphaned commit on a fork somewhere?

I have ended up instead on the dotnet-vnext branch at https://github.com/martincostello/dependabot-helper/commit/398cdd180af17d6afe8db049d80e32c1048750db and then I undid the edit to `tests/DependabotHelper.Tests/DependabotHelper.Tests.csproj` that disabled PGO. 

Looks like I also need to install Powershell 7 and npm?

**Comment by martincostello** [comment]
That commit was in a PR that squash merged into that branch once the tests passed.

You'll need node/npm to build the JS for the UI and PowerShell 7 to use the build script, but if you don't want to use that you could run dotnet test manually with the equivalent command from it which would be something like this:

`dotnet test --project ./tests/DependabotHelper.Tests --filter Can_Configure_Repositories`

**Comment by martincostello** [comment]
I found a 5th repo seemingly affected by this, but with a different failure mode, but ultimately an unexpected failure in the same area of the Playwright code:

```
  Error Message:
   Microsoft.Playwright.PlaywrightException : Browser has been closed
  Stack Trace:
     at Microsoft.Playwright.Transport.Connection.InnerSendMessageToServerAsync[T](String guid, String method, Dictionary`2 dictionary, Boolean keepNulls) in /_/src/Playwright/Transport/Connection.cs:line 131
   at Microsoft.Playwright.Transport.Connection.WrapApiCallAsync[T](Func`1 action, Boolean isInternal) in /_/src/Playwright/Transport/Connection.cs:line 513
   at Microsoft.Playwright.Core.BrowserContext.CloseAsync() in /_/src/Playwright/Core/BrowserContext.cs:line 261
```

Disabling Dynamic PGO also solved that issue. This _might_ be easier for you to repro the issue with:

1. Checkout https://github.com/martincostello/dotnet-playwright-tests/pull/128/commits/7239309ff5b1f7431bef34269d17effaf810f537
2. Run `dotnet test ./PlaywrightTests --configuration Release`

**Comment by AndyAyersMS** [comment]
I can repro the original failure. Looks like at 

https://github.com/microsoft/playwright-dotnet/blob/ca7b02ac910e573666f48d502bfabb1c17639e68/src/Playwright/Transport/Connection.cs#L150

The `ApiZone.Value[0]` array entry is `null`. 

`ApiZone` is an async local, looks like it should be getting set up by this code

https://github.com/microsoft/playwright-dotnet/blob/ca7b02ac910e573666f48d502bfabb1c17639e68/src/Playwright/Transport/Connection.cs#L465-L519

And upstack we are sitting just below the code that should have initialized that array entry:

<img width="671" alt="image" src="https://github.com/dotnet/runtime/assets/10121823/b35cb26f-eecd-43ca-ad28-1e5951ea2d80">

and we are indeed in tier1 optimized code for that method:

```
0:019> !ip2md 00007ffb`4ba85722 
MethodDesc:   00007ffa99d8d048
Method Name:          Microsoft.Playwright.Transport.Connection+<WrapApiCallAsync>d__49`1[[System.__Canon, System.Private.CoreLib]].MoveNext()
Class:                00007ffa99d92fa0
MethodTable:          00007ffa99d8d0e0
mdToken:              00000000060016D2
Module:               00007ffa96ff9ee0
IsJitted:             yes
Current CodeAddr:     00007ffb4ba84ba0
Version History:
  ILCodeVersion:      0000000000000000
  ReJIT ID:           0
  IL Addr:            000001dbfc7c807c
     CodeAddr:           00007ffb4b9a8240  (QuickJitted + Instrumented)
     NativeCodeVersion:  0000019B03E3DC30
     CodeAddr:           00007ffb4ba84ba0  (OptimizedTier1)
     NativeCodeVersion:  0000019B02948160
     CodeAddr:           00007ffa99c47df0  (QuickJitted)
     NativeCodeVersion:  0000000000000000
Source file:  /_/src/Playwright/Transport/Connection.cs @ 513
```
So potentially there is a codegen bug in that method

**Comment by AndyAyersMS** [comment]
I wonder if the bug is that `WrapApiCallAsync` expects to be able to walk the stack and find certain key method names, and with PGO these methods may now be inlined and so this stack inspection no longer works properly?

The actual (non-sanitized) stack is something like
```
[0x26]   Microsoft_Playwright!Microsoft.Playwright.Transport.Connection.<WrapApiCallAsync>d__49<Microsoft.Playwright.Transport.Channels.ElementHandleChannel>.MoveNext+0xb82   0x53ab53e410   0x7ffb4bac5147   
[0x27]   System_Private_CoreLib!System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[MartinCostello.DependabotHelper.GitHubService+<CacheGetOrCreateAsync>d__28`1[[System.__Canon, System.Private.CoreLib]], DependabotHelper]](<CacheGetOrCreateAsync>d__28`1<System.__Canon> ByRef)+0xb19624b7   0x53ab53e540   0x7ffb4bac4a37   
[0x28]   Microsoft_Playwright!Microsoft.Playwright.Core.ElementHandle.<QuerySelectorAsync>d__25.MoveNext+0x147   0x53ab53e5c0   0x7ffb4bad1a1f   
[0x29]   System_Private_CoreLib!System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[MartinCostello.DependabotHelper.GitHubService+<CacheGetOrCreateAsync>d__28`1[[System.__Canon, System.Private.CoreLib]], DependabotHelper]](<CacheGetOrCreateAsync>d__28`1<System.__Canon> ByRef)+0xb196ed8f   0x53ab53e680   0x7ffb4bde9642   

```
and it seems possible that `lastInternalApiName` never gets set to anything non-null, which would explain the failure.

**Comment by AndyAyersMS** [comment]
For example, we see `WrapApiCallAsync` get inlined into `Microsoft.Playwright.Core.ElementHandle+<QuerySelectorAsync>d__25:MoveNext():this`:
```
Inlines into 0600184D [via ExtendedDefaultPolicy] Microsoft.Playwright.Core.ElementHandle+<QuerySelectorAsync>d__25:MoveNext():this:
  [INL01 IL=0029 TR=000077 06001055] [INLINED: call site: profitable inline] Microsoft.Playwright.Transport.Channels.ElementHandleChannel:QuerySelectorAsync(System.String):System.Threading.Tasks.Task`1[Microsoft.Playwright.Transport.Channels.ElementHandleChannel]:this
    [INL02 IL=0001 TR=000132 06001048] [INLINED: callee: below ALWAYS_INLINE size] Microsoft.Playwright.Transport.Channels.ChannelBase:get_Connection():Microsoft.Playwright.Transport.Connection:this
    [INL03 IL=0007 TR=000135 06001047] [INLINED: callee: below ALWAYS_INLINE size] Microsoft.Playwright.Transport.Channels.ChannelBase:get_Guid():System.String:this
    [INL04 IL=0017 TR=000146 060081CC] [INLINED: callee: below ALWAYS_INLINE size] System.Collections.Generic.Dictionary`2[System.__Canon,System.__Canon]:.ctor():this
      [INL00 IL=0003 TR=000174 060081CF] [FAILED: callee: noinline per IL/cached result] System.Collections.Generic.Dictionary`2[System.__Canon,System.__Canon]:.ctor(int,System.Collections.Generic.IEqualityComparer`1[System.__Canon]):this
    [INL05 IL=0029 TR=000154 060081DF] [INLINED: callee: below ALWAYS_INLINE size] System.Collections.Generic.Dictionary`2[System.__Canon,System.__Canon]:set_Item(System.__Canon,System.__Canon):this
      [INL00 IL=0004 TR=000178 060081ED] [FAILED: callee: noinline per IL/cached result] System.Collections.Generic.Dictionary`2[System.__Canon,System.__Canon]:TryInsert(System.__Canon,System.__Canon,ubyte):bool:this
    [INL06 IL=0035 TR=000156 06000D0C] [INLINED: call site: profitable inline] Microsoft.Playwright.Transport.Connection:SendMessageToServerAsync[System.__Canon](System.String,System.String,System.Collections.Generic.Dictionary`2[System.String,System.Object],bool):System.Threading.Tasks.Task`1[System.__Canon]:this
      [INL07 IL=0000 TR=000190 060016C8] [INLINED: callee: below ALWAYS_INLINE size] Microsoft.Playwright.Transport.Connection+<>c__DisplayClass38_0`1[System.__Canon]:.ctor():this
        [INL08 IL=0001 TR=000239 0600063F] [INLINED: callee: below ALWAYS_INLINE size] System.Object:.ctor():this
      [INL09 IL=0050 TR=000224 06000635] [INLINED: call site: profitable inline] System.MulticastDelegate:CtorClosed(System.Object,long):this
        [INL00 IL=0003 TR=000254 06000634] [FAILED: callee: does not return] System.MulticastDelegate:ThrowNullThisInDelegateToInstance()
      [INL10 IL=0056 TR=000227 06000D17] [INLINED: call site: profitable inline] Microsoft.Playwright.Transport.Connection:WrapApiCallAsync[System.__Canon](System.Func`1[System.__Canon],bool):System.Threading.Tasks.Task`1[System.__Canon]:this
        [INL11 IL=0002 TR=000258 060061F6] [INLINED: callee: below ALWAYS_INLINE size] System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]:Create():System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]
        [INL12 IL=0053 TR=000285 060061F7] [INLINED: callee: aggressive inline attribute] System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]:Start[Microsoft.Playwright.Transport.Connection+<WrapApiCallAsync>d__49`1[System.__Canon]](byref):this
          [INL00 IL=0001 TR=000300 060061E1] [FAILED: callee: has exception handling] System.Runtime.CompilerServices.AsyncMethodBuilderCore:Start[Microsoft.Playwright.Transport.Connection+<WrapApiCallAsync>d__49`1[System.__Canon]](byref)
        [INL13 IL=0065 TR=000289 06006200] [INLINED: callee: aggressive inline attribute] System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]:get_Task():System.Threading.Tasks.Task`1[System.__Canon]:this
          [INL00 IL=0011 TR=000321 06006201] [FAILED: callee: noinline per IL/cached result] System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]:InitializeTaskAsPromise():System.Threading.Tasks.Task`1[System.__Canon]:this
  [INL14 IL=0035 TR=000080 06003A77] [INLINED: callee: below ALWAYS_INLINE size] System.Threading.Tasks.Task`1[System.__Canon]:ConfigureAwait(bool):System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1[System.__Canon]:this
    [INL15 IL=0002 TR=000333 06006364] [INLINED: callee: below ALWAYS_INLINE size] System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1[System.__Canon]:.ctor(System.Threading.Tasks.Task`1[System.__Canon],bool):this
      [INL16 IL=0003 TR=000349 06006366] [INLINED: callee: below ALWAYS_INLINE size] System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[System.__Canon]:.ctor(System.Threading.Tasks.Task`1[System.__Canon],bool):this
  [INL17 IL=0044 TR=000085 06006365] [INLINED: callee: below ALWAYS_INLINE size] System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1[System.__Canon]:GetAwaiter():System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[System.__Canon]:this
  [INL18 IL=0053 TR=000091 06006367] [INLINED: callee: below ALWAYS_INLINE size] System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[System.__Canon]:get_IsCompleted():bool:this
    [INL19 IL=0006 TR=000377 06003B30] [INLINED: callee: below ALWAYS_INLINE size] System.Threading.Tasks.Task:get_IsCompleted():bool:this
      [INL20 IL=0010 TR=000386 06003B31] [INLINED: callee: below ALWAYS_INLINE size] System.Threading.Tasks.Task:IsCompletedMethod(int):bool
  [INL21 IL=0086 TR=000114 060061FB] [INLINED: callee: aggressive inline attribute] System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]:AwaitUnsafeOnCompleted[System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[System.__Canon],Microsoft.Playwright.Core.ElementHandle+<QuerySelectorAsync>d__25](byref,byref):this
    [INL22 IL=0008 TR=000399 060061FC] [INLINED: callee: aggressive inline attribute] System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]:AwaitUnsafeOnCompleted[System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[System.__Canon],Microsoft.Playwright.Core.ElementHandle+<QuerySelectorAsync>d__25](byref,byref,byref)
      [INL00 IL=0002 TR=000405 060061FE] [FAILED: callee: too many il bytes] System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]:GetStateMachineBox[Microsoft.Playwright.Core.ElementHandle+<QuerySelectorAsync>d__25](byref,byref):System.Runtime.CompilerServices.IAsyncStateMachineBox
      [INL00 IL=0010 TR=000411 060061FD] [FAILED: callee: has exception handling] System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]:AwaitUnsafeOnCompleted[System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[System.__Canon]](byref,System.Runtime.CompilerServices.IAsyncStateMachineBox)
  [INL23 IL=0124 TR=000038 0600636A] [INLINED: call site: profitable inline] System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[System.__Canon]:GetResult():System.__Canon:this
    [INL24 IL=0006 TR=000422 0600634F] [INLINED: callee: below ALWAYS_INLINE size] System.Runtime.CompilerServices.TaskAwaiter:ValidateEnd(System.Threading.Tasks.Task)
      [INL25 IL=0001 TR=000430 06003B15] [INLINED: callee: aggressive inline attribute] System.Threading.Tasks.Task:get_IsWaitNotificationEnabledOrNotRanToCompletion():bool:this
      [INL00 IL=0009 TR=000436 06006350] [FAILED: call site: unprofitable inline] System.Runtime.CompilerServices.TaskAwaiter:HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task)
    [INL26 IL=0017 TR=000426 06003A72] [INLINED: callee: below ALWAYS_INLINE size] System.Threading.Tasks.Task`1[System.__Canon]:get_ResultOnSuccess():System.__Canon:this
  [INL27 IL=0137 TR=000048 06001051] [INLINED: callee: below ALWAYS_INLINE size] Microsoft.Playwright.Transport.Channels.ElementHandleChannel:get_Object():Microsoft.Playwright.Core.ElementHandle:this
  [INL00 IL=0163 TR=000129 06006205] [FAILED: call site: within catch region] System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]:SetException(System.Exception):this
  [INL28 IL=0185 TR=000064 06006203] [INLINED: call site: profitable inline] System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]:SetResult(System.__Canon):this
    [INL29 IL=0010 TR=000471 06003BAD] [INLINED: callee: aggressive inline attribute] System.Threading.Tasks.Task:FromResult[System.__Canon](System.__Canon):System.Threading.Tasks.Task`1[System.__Canon]
      [INL30 IL=0119 TR=000504 06006199] [INLINED: callee: below ALWAYS_INLINE size] System.Runtime.CompilerServices.RuntimeHelpers:IsReferenceOrContainsReferences[System.__Canon]():bool
      [INL31 IL=0193 TR=000516 06003A5F] [INLINED: call site: profitable inline] System.Threading.Tasks.Task`1[System.__Canon]:.ctor(System.__Canon):this
        [INL32 IL=0012 TR=000571 06003AFE] [INLINED: call site: profitable inline] System.Threading.Tasks.Task:.ctor(bool,int,System.Threading.CancellationToken):this
          [INL33 IL=0001 TR=000581 0600063F] [INLINED: callee: below ALWAYS_INLINE size] System.Object:.ctor():this
    [INL34 IL=0028 TR=000467 06006204] [INLINED: call site: profitable inline] System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.__Canon]:SetExistingTaskResult(System.Threading.Tasks.Task`1[System.__Canon],System.__Canon)
      [INL35 IL=0005 TR=000599 06007B87] [INLINED: callee: below ALWAYS_INLINE size] System.Diagnostics.Tracing.EventSource:IsEnabled():bool:this
      [INL00 IL=0018 TR=000615 06003B24] [FAILED: call site: unprofitable inline] System.Threading.Tasks.Task:get_Id():int:this
      [INL00 IL=0024 TR=000621 06003D48] [FAILED: call site: unprofitable inline] System.Threading.Tasks.TplEventSource:TraceOperationEnd(int,int):this
      [INL36 IL=0031 TR=000605 06003A6F] [INLINED: call site: profitable inline] System.Threading.Tasks.Task`1[System.__Canon]:TrySetResult(System.__Canon):bool:this
        [INL37 IL=0013 TR=000637 06003B0F] [INLINED: call site: profitable inline] System.Threading.Tasks.Task:AtomicStateUpdate(int,int):bool:this
          [INL00 IL=0035 TR=000718 06003B10] [FAILED: call site: unprofitable inline] System.Threading.Tasks.Task:AtomicStateUpdateSlow(int,int):bool:this
        [INL00 IL=0064 TR=000676 06003B52] [FAILED: call site: unprofitable inline] System.Threading.Tasks.Task:NotifyParentIfPotentiallyAttachedTask():this
        [INL00 IL=0070 TR=000678 06003BD6] [FAILED: call site: unprofitable inline] System.Threading.Tasks.Task+ContingentProperties:SetCompleted():this
        [INL38 IL=0076 TR=000671 06003B7C] [INLINED: call site: profitable inline] System.Threading.Tasks.Task:FinishContinuations():this
          [INL00 IL=0011 TR=000728 06003506] [FAILED: callee: noinline per IL/cached result] System.Threading.Interlocked:Exchange(byref,System.Object):System.Object
          [INL00 IL=0022 TR=000737 06003B7D] [FAILED: callee: noinline per IL/cached result] System.Threading.Tasks.Task:RunContinuations(System.Object):this
      [INL00 IL=0040 TR=000611 0600218F] [FAILED: callee: does not return] System.ThrowHelper:ThrowInvalidOperationException(int)

**Comment by AndyAyersMS** [comment]
Opened a bug in playwright. Odd that this isn't auto-linking in some browser view, so will add the link here explicitly.

https://github.com/microsoft/playwright-dotnet/issues/2617

**Comment by AndyAyersMS** [comment]
The issue in playwright is fixed in their sources -- waiting on info on a release with the fix.

**Comment by AndyAyersMS** [comment]
Thank you for verifying this fix @martincostello. Fix is in `1.36.0-beta-1` and later.

I am going to close this as resolved.
